lib/hls/hls_parser.js

```diff
@@ -1966,6 +2034,79 @@ shaka.hls.HlsParser = class {
       references.push(reference);
     }
 
+    // If some segments have sync times, but not all, extrapolate the sync
+    // times of the ones with none.
+    const someSyncTime = references.some((ref) => ref.syncTime != null);
+    if (someSyncTime) {
+      for (let i = 0; i < references.length; i++) {
+        const reference = references[i];
+        if (reference.syncTime != null) {
+          // No need to extrapolate.
+          continue;
+        }
+        // Find the nearest segment with syncTime, in either direction.
+        // This looks forward and backward simultaneously, keeping track of what
+        // to offset the syncTime it finds by as it goes.
+        let forwardAdd = 0;
+        let forwardI = i;
+        /**
+         * Look forwards one reference at a time, summing all durations as we
+         * go, until we find a reference with a syncTime to use as a basis.
+         * This DOES count the original reference.
```

Comment:
but DOESN'T count the first reference with a syncTime (since we approach it from behind).

Reply:
Done.